src/Hedgehog/Gen.fs

```diff
@@ -1,57 +1,73 @@
 ﻿namespace Hedgehog
 
 open System
+open Hedgehog.Numeric
 
 /// A generator for values and shrink trees of type 'a.
 [<Struct>]
 type Gen<'a> =
-    | Gen of Random<Tree<'a>>
+    | Gen of (Seed -> Size -> Tree<'a>)
 
 module Gen =
 
-    let ofRandom (r : Random<Tree<'a>>) : Gen<'a> =
-        Gen r
+    let private unsafeRun (seed : Seed) (size : Size) (Gen(g) : Gen<'a>) : Tree<'a> =
+        g seed size
 
-    let toRandom (Gen r : Gen<'a>) : Random<Tree<'a>> =
-        r
+    let run (seed : Seed) (size : Size) (g : Gen<'a>) : Tree<'a> =
+        unsafeRun seed (max 1 size) g
+
+    let extract (seed : Seed) (size : Size) (g : Gen<'a>) : 'a =
```

Comment:
Do we really want to expose this? Can't we just inline it below in that test?

Reply:
This probably doesn't need to be exposed, I was thinking of the comonad function `extract`, which we might be able to add in earnest later. This signature doesn't follow the usual signature `extract :: m a -> a` and I don't think we have any `extend` concept to make it a proper comonad. What are your thoughts on that @moodmosaic?

Reply:
I love @adam-becker's idea to implement the randomness as a comonad, so let's not expose any additional parts of the current randomness API lest they become breaking changes later.

Reply:
>the usual signature `extract :: m a -> a`

`m` should be `w` (upside down *m*!) since it's a comonad 👀 I haven't thought about using comonads in generators and PRNGs.